plugins/scaffolder-node/src/actions/createScaffolderAction.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { JsonObject } from '@backstage/types';
+import { Schema } from 'jsonschema';
+import { z } from 'zod';
+import zodToJsonSchema from 'zod-to-json-schema';
+import { ScaffolderAction, ScaffolderActionContext } from './types';
+
+/** @public */
+export type ScaffolderExample = {
+  description: string;
+  example: string;
+};
+
+/** @public */
+export type ScaffolderActionOptions<
```

Comment:
So I think as discussed we want to make this an override to the original function rather than creating something separate from the get-go.

Also thinking that we can trim down and simplify these types a lot if we remove the ability to specify the actions as `JSONSchema` and only rely on zod instead.

I think that we wanna head down this route:

```ts
createTemplateAction({
  id: 'test',
  description,
  examples,
  schema: {
    input: {
      repoUrl: (z) => z.string(),
      token: (z) => z.string().optional(),
    },
    output: {
      something: (z) => z.string(),
    },
  },
  async handler(ctx) {
    ...
  }
});
```

By removing the `JSONSchema` option, it simplifies both the implementation and also consumption because end users then don't need to worry about having to type the input and output types twice when providing JSONSchema like this:

https://github.com/backstage/backstage/blob/8743beff4d843cc0fe0896a04407369f2db5c902/plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/plain.ts#L42-L46